drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+    data->ambient_rate |= raw.ambient_count_rate_mcps_sd0_lo;
+
+    /* convert values from 9.7 to 16.16 */
+    data->signal_rate  <<= 9;
+    data->ambient_rate <<= 9;
+
+    #endif /* !MODULE_VL53L1X_BASIC */
+
+    /* restart measurement */
+    EXEC_RET(_start(dev));
+
+    #endif /* MODULE_VL53L1X_ST_API */
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_read_mm(vl53l1x_t *dev, int16_t *mm)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(mm != NULL);
+    DEBUG_DEV("mm=%p", dev, mm);
+
+    vl53l1x_data_t data;
+    EXEC_RET(vl53l1x_read_data (dev, &data));
+    *mm = data.distance;
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_int_config(vl53l1x_t *dev, vl53l1x_int_config_t* cfg,
+                                       void (*isr)(void *),
+                                       void *isr_arg)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(dev->params.pin_int != GPIO_UNDEF);
+
+    DEBUG_DEV("cfg=%p isr=%p isr_arg=%p", dev, cfg, isr, isr_arg);
+
+    if (!dev->int_init) {
+        dev->int_init = true;
+        gpio_init_int(dev->params.pin_int, GPIO_IN_PU, GPIO_FALLING,
+                      isr, isr_arg);
+    }
+
+    #if MODULE_VL53L1X_ST_API
+    ASSERT_PARAM(cfg != NULL);
+
+    EXEC_RET(VL53L1_StopMeasurement((VL53L1_Dev_t*)dev));
+
+    VL53L1_DetectionConfig_t dcfg;
+    dcfg.DetectionMode = cfg->mode;
+    dcfg.Distance.CrossMode = cfg->dist_mode;
+    dcfg.Distance.High = cfg->dist_high;
+    dcfg.Distance.Low = cfg->dist_low;
+    dcfg.Rate.CrossMode = cfg->rate_mode;
+    dcfg.Rate.High = cfg->rate_high;
+    dcfg.Rate.Low = cfg->rate_low;
+    dcfg.IntrNoTarget = 0;
+
+    EXEC_RET(VL53L1_SetThresholdConfig((VL53L1_Dev_t*)dev, &dcfg));
+    EXEC_RET(VL53L1_StartMeasurement((VL53L1_Dev_t*)dev));
+
+    #else
+    DEBUG_DEV("configuration is not used", dev);
+    #endif
+
+    return VL53L1X_OK;
+}
+
+#if !MODULE_VL53L1X_BASIC
+
+int vl53l1x_power_down(const vl53l1x_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    if (dev->params.pin_shutdown == GPIO_UNDEF) {
+        DEBUG_DEV("Pin connected to sensor's XSHUT pin not defined", dev);
+        return VL53L1X_ERROR_NO_PIN;
+    }
+
+    gpio_clear(dev->params.pin_shutdown);
+    return VL53L1X_OK;
+}
+
+int vl53l1x_power_up(vl53l1x_t *dev)
+{
+    ASSERT_PARAM(dev != NULL);
+    DEBUG_DEV("", dev);
+
+    if (dev->params.pin_shutdown == GPIO_UNDEF) {
+        DEBUG_DEV("Pin connected to sensor's XSHUT pin not defined", dev);
+        return VL53L1X_ERROR_NO_PIN;
+    }
+
+    gpio_set(dev->params.pin_shutdown);
+
+    /* init the sensor and start measurement */
+    _init(dev);
+
+    return VL53L1X_OK;
+}
+
+int vl53l1x_get_timing_budget(const vl53l1x_t *dev, uint32_t *budget)
+{
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(budget != NULL);
+    DEBUG_DEV("budget=%p", dev, budget);
+
+    #if MODULE_VL53L1X_ST_API
+    return VL53L1_GetMeasurementTimingBudgetMicroSeconds((VL53L1_Dev_t*)dev,
```

Comment:
```suggestion
    return VL53L1_GetMeasurementTimingBudgetMicroSeconds(&dev->dev,
```

Btw: Are the IT programmers payed by character count? :rofl: